models/certs.go

```diff
@@ -32,3 +34,51 @@ type CertInfo struct {
 	ValidTo   string    `json:"ValidTo"`
```

Comment:
**Code Review Report**

**Code Styling**

* The code is written in the correct format, but it would be better to use Go's standard package for formatting, `gofmt`, to ensure consistency in formatting throughout the codebase.

**Error Handling**

* In the `NewCSRConfigModel` and `NewCertInfoModel` functions, the `types.StringValue` function is called without checking if the input string is nil. It would be more robust to check if the string is not nil before calling the function.

**Code Snippet:**
```go
func NewCSRConfigModel(input CSRConfig) CSRConfigModel {
    if input.DistinguishedName == nil {
        return CSRConfigModel{}
    }
    return CSRConfigModel{
        // ...
    }
}
```
**Type Conversions**

* The `NewCSRConfigModel` and `NewCertInfoModel` functions convert `string` types to `types.String` types. It would be more efficient and clear to use the `types.String` type directly in the struct definition, removing the need for these conversions.

**Code Snippet:**
```go
type CSRConfigModel struct {
    DistinguishedName types.String `tfsdk:"distinguished_name"`
    // ...
}
```
**Unused Imports**

* The `github.com/hashicorp/terraform-plugin-framework/types` import is used, but the `types.String` type is already defined in the same package. The unused import can be removed.

**Code Snippet:**
```go
import "github.com/hashicorp/terraform-plugin-framework"
```
**Terraform-Provider Standards**

* The code does not comply with the Hasicorp Terraform provider standards. Specifically, the `tfsdk:""` tags should be used for the struct fields, and the `description` field should be provided for each struct.

**Code Snippet:**
```go
type CSRConfigModel struct {
    DistinguishedName types.String `tfsdk:"distinguished_name" description:"The distinguished name of the certificate"`
    // ...
}
```